test/cql-pytest/test_secondary_index.py

```diff
@@ -241,3 +241,23 @@ def test_filter_cluster_key(cql, test_keyspace):
         stmt = SimpleStatement(f"SELECT c1, c2 FROM {table} WHERE c1 = 1 and c2 = 1 ALLOW FILTERING")
         rows = cql.execute(stmt)
         assert_rows(rows, [1, 1])
+
+# Reproduces #9085.  Scylla-only because Cassandra uses indices differently.
```

Comment:
I don't understand this test. The example in 9085 shows wrong output. Why don't you just test that? It seems you are just testing that ALLOW FILTERING is needed. That's perhaps a good test for test_allow_filtering.py, but here it would be better to add ALLOW FILTERING and test that the **result** is correct - and should be correct also in Cassandra. Making a correctness test scylla_only doesn't make sense.

Reply:
> The example in 9085 shows wrong output. Why don't you just test that?

I do, below.  But the root cause of #9085 is using the index when we shouldn't.  The fix, then, is to ignore the index.  Which, unfortunately, lands us exactly in the #6200 scenario. :(  We cannot obtain correct results until we fix #6200.  We can, however, ensure that the index isn't used.  I'm happy to move it to test_allow_filtering.py.

Reply:
I don't know if it should move to test_allow_filtering.py. That file is just for testing whether or not scylla requires "ALLOW FILTERING" on specific queries - it's not about internal implementation details but really visible ALLOW-FILTERING-requirement differences between Scylla and Cassandra. Is this the case here?

But more importantly, I still don't understand why this test is "scylla-only". Is it because we consider Scylla's behavior correct and Cassandra's behavior incorrect? If this is the case, we should mark it cassandra-bug, not scylla-only (but should only do this after careful thinking). Or, alternatively, maybe you consider Scylla's behavior is wrong here but you want to keep it **for now**? In that case, the test should pass on Cassandra's behavior - not Scylla's behavior - and should xfail on Scylla. You can see a few tests like that in `test_allow_filtering.py`.

So in both cases I described using `scylla_only` for such test is wrong. `scylla_only` makes sense mostly for testing Scylla-only features, and sometimes also for cases where we deliberately decided to diverge from Cassandra - not temporarily, not until we fix the code - but a deliberate decision that Cassandra did something badly and we can do it better and will never want to go back to their behavior. Is this the case here?

Reply:
> But more importantly, I still don't understand why this test is "scylla-only". Is it because we consider Scylla's behavior correct and Cassandra's behavior incorrect? 

We definitely consider Scylla's behaviour correct.  And if Cassandra's indices were identical to Scylla's indices, we'd consider Cassandra's behaviour incorrect.  But I don't know enough about Cassandra's indices; for all I know, they may be sufficiently powerful to eliminate the filtering step here.